Firebase/Core/FIRLogger.m

```diff
@@ -86,170 +64,53 @@
 
 void FIRLoggerInitializeASL() {
   dispatch_once(&sFIRLoggerOnceToken, ^{
-    NSInteger majorOSVersion = [[FIRAppEnvironmentUtil systemVersion] integerValue];
-    uint32_t aslOptions = ASL_OPT_STDERR;
-#if TARGET_OS_SIMULATOR
-    // The iOS 11 simulator doesn't need the ASL_OPT_STDERR flag.
-    if (majorOSVersion >= 11) {
-      aslOptions = 0;
-    }
-#else
-    // Devices running iOS 10 or higher don't need the ASL_OPT_STDERR flag.
-    if (majorOSVersion >= 10) {
-      aslOptions = 0;
-    }
-#endif  // TARGET_OS_SIMULATOR
-
-    // Override the aslOptions to ASL_OPT_STDERR if the override argument is passed in.
     NSArray *arguments = [NSProcessInfo processInfo].arguments;
-    if ([arguments containsObject:kFIRLoggerForceSDTERRApplicationArgument]) {
-      aslOptions = ASL_OPT_STDERR;
-    }
-
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"  // asl is deprecated
-    // Initialize the ASL client handle.
-    sFIRLoggerClient = asl_open(NULL, kFIRLoggerASLClientFacilityName, aslOptions);
-
-    // Set the filter used by system/device log. Initialize in default mode.
-    asl_set_filter(sFIRLoggerClient, ASL_FILTER_MASK_UPTO(ASL_LEVEL_NOTICE));
-    sFIRLoggerDebugMode = NO;
-    sFIRAnalyticsDebugMode = NO;
-    sFIRLoggerMaximumLevel = FIRLoggerLevelNotice;
-
-    // Use the standard NSUserDefaults if it hasn't been explicitly set.
-    if (sFIRLoggerUserDefaults == nil) {
-      sFIRLoggerUserDefaults = [NSUserDefaults standardUserDefaults];
-    }
-
-    BOOL debugMode = [sFIRLoggerUserDefaults boolForKey:kFIRPersistedDebugModeKey];
-    if ([arguments containsObject:kFIRDisableDebugModeApplicationArgument]) {  // Default mode
-      [sFIRLoggerUserDefaults removeObjectForKey:kFIRPersistedDebugModeKey];
-    } else if ([arguments containsObject:kFIREnableDebugModeApplicationArgument] ||
-               debugMode) {  // Debug mode
-      [sFIRLoggerUserDefaults setBool:YES forKey:kFIRPersistedDebugModeKey];
-      asl_set_filter(sFIRLoggerClient, ASL_FILTER_MASK_UPTO(ASL_LEVEL_DEBUG));
-      sFIRLoggerDebugMode = YES;
-    }
-
-    // We should disable debug mode if we are running from App Store.
-    if (sFIRLoggerDebugMode && [FIRAppEnvironmentUtil isFromAppStore]) {
-      sFIRLoggerDebugMode = NO;
-    }
-
-    sFIRClientQueue = dispatch_queue_create("FIRLoggingClientQueue", DISPATCH_QUEUE_SERIAL);
-    dispatch_set_target_queue(sFIRClientQueue,
-                              dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0));
-
-#ifdef DEBUG
-    sMessageCodeRegex =
-        [NSRegularExpression regularExpressionWithPattern:kMessageCodePattern options:0 error:NULL];
-#endif
+    GULLoggerInitializeASL([arguments containsObject:kFIRDisableDebugModeApplicationArgument],
```

Comment:
Was there an API review for this?

If not:
  * Let's drop "ASL", which is deprecated. We should eventually move to `os_log`.
  * Is there any way we can avoid passing three booleans to this? It's a terrible interface.
  * Is there a way to translate debug modes into a maximum log level instead so that GULLogger doesn't even have to be aware of this wart?

In particular it seems like if argument processing is to be done here, then we should completely figure out if we want debug mode from the arguments locally. Passing two booleans that seem like logical inverses of each other seems wrong.

Reply:
+1 to moving to `os_log` but it's only available on iOS 10+, so we'll have to keep `ASL` around until we drop support for iOS 8 and 9 (will be a while).